solr/core/src/java/org/apache/solr/request/json/RequestUtil.java

```diff
@@ -241,13 +243,16 @@ public static void processParams(
           if (queriesJsonObj instanceof Map && queriesJsonObj != null) {
             @SuppressWarnings("unchecked")
             final Map<String, Object> queriesAsMap = (Map<String, Object>) queriesJsonObj;
+            ArrayList<String> queryKeys = new ArrayList<>();
             for (Map.Entry<String, Object> queryJsonProperty : queriesAsMap.entrySet()) {
               out = queryJsonProperty.getKey();
+              queryKeys.add(out);
               arr = true;
               isQuery = true;
               convertJsonPropertyToLocalParams(
                   newMap, jsonQueryConverter, queryJsonProperty, out, isQuery, arr);
             }
+            newMap.put(COMBINER_KEYS, queryKeys.toArray(new String[queryKeys.size()]));
```

Comment:
```suggestion
            newMap.put(COMBINER_KEYS, queryKeys.toArray(new String[0]));
```
We standardized on this recently; surprisingly a 0 helps perf